File: Data-Structures-and-Algorithms-using-Swift/IntroductoryChapters/Analysis.swift
import Foundation

func fun1(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		m += 1
		i+=1
	}
	return m
}

func fun2(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		var j : Int = 0
		while j < n{
			m += 1
			j+=1 
		}
		i+=1
	}
	return m
}

func fun3(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		var j : Int = 0
		while j < n {
			var k = 0
			while k < n{
				m += 1
				k += 1
			}
			j+=1
		}
		i+=1
	}
	return m
}

func fun4(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		var j = i
		while j < n {
			var k = j + 1
			while k < n {
				m += 1
				k += 1
			}
			j+=1
		}
		i+=1
	}
	return m
}

func fun5(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n {
		var j : Int = 0 
		while j < i {
			m += 1
			j += 1
		}
		i += 1
	}
	return m
}

func fun6(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		var j = i
		while j > 0 {
			m += 1
			j -= 1
		}
		i+=1
	}
	return m
}

func fun7(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = n
	while i > 0 {
		var j : Int = 0 
		while j < i {  
			m += 1
			j+=1
		}
		i /= 2
	}
	return m
}

func fun8(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 1
	while i <= n {
		var j : Int = 0
		while j <= i {
			m += 1
			j+=1
		}
		i *= 2
	}
	return m
}

func fun9(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 1
	while i < n {
		m += 1
		i = i * 2
	}
	return m
}

func fun10(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = n
	while i > 0 {
		m += 1
		i = i / 2
	}
	return m
}

func fun11(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		var j : Int = 0
		while j < n {
			m += 1
			j+=1
		}
		i+=1
	}
	i = 0 
	while i < n { 
		var k = 0
		while k < n {
			m += 1
			k += 1
		}
		i+=1
	}
	return m
}

func fun12(_ n : Int) -> Int {
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		let sq = sqrt(Double(n))
		var j : Int = 0
		while j < Int(sq) {
			m += 1
			j += 1
		}
		i+=1
	}

	return m
}

func fun13(_ n : Int) -> Int {
	var j = 0
	var m : Int = 0
	var i : Int = 0 
	while i < n { 
		while j < n {
			m += 1
			j+=1
		}
		i+=1
	}
	return m
}

print("N = 100, Number of instructions in O(n):: \(fun1(100))")
print("N = 100, Number of instructions in O(n^2):: \(fun2(100))")
print("N = 100, Number of instructions in O(n^3):: \(fun3(100))")
print("N = 100, Number of instructions in O(n^3):: \(fun4(100))")
print("N = 100, Number of instructions in O(n^2):: \(fun5(100))")
print("N = 100, Number of instructions in O(n^2):: \(fun6(100))")
print("N = 100, Number of instructions in O(n):: \(fun7(100))")
print("N = 100, Number of instructions in O(n):: \(fun8(100))")
print("N = 100, Number of instructions in O(log(n)):: \(fun9(100))")
print("N = 100, Number of instructions in O(log(n)):: \(fun10(100))")
print("N = 100, Number of instructions in O(n^2):: \(fun11(100))")
print("N = 100, Number of instructions in O(n^(3/2)):: \(fun12(100))")
print("N = 100, Number of instructions in O(n):: \(fun13(100))")

/*
N = 100, Number of instructions in O(n):: 100
N = 100, Number of instructions in O(n^2):: 10000
N = 100, Number of instructions in O(n^3):: 1000000
N = 100, Number of instructions in O(n^3):: 166650
N = 100, Number of instructions in O(n^2):: 4950
N = 100, Number of instructions in O(n^2):: 4950
N = 100, Number of instructions in O(n):: 197
N = 100, Number of instructions in O(n):: 134
N = 100, Number of instructions in O(log(n)):: 7
N = 100, Number of instructions in O(log(n)):: 7
N = 100, Number of instructions in O(n^2):: 20000
N = 100, Number of instructions in O(n^(3/2)):: 1000
N = 100, Number of instructions in O(n):: 100
*/

File: Data-Structures-and-Algorithms-using-PHP/IntroductoryChapters/Analysis.php
<?php
Function Fun1($n) {
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	$m += 1;
	return $m;
}

Function Fun2($n) {
	$i = null;
	$j = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $n); ++ $j)
	$m += 1;
	return $m;
}

Function Fun3($n)
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $i); ++ $j)
	$m += 1;
	return $m;
}

Function Fun4($n)
{
	$i = null;
	$m = 0;
	$i = 1;
	while ( ($i < $n) ) {
		$m += 1;
		$i = ($i * 2);
	}
	return $m;
}

Function Fun5($n)
{
	$i = null;
	$m = 0;
	$i = $n;
	while ( $i >= 1 ) {
		$m += 1;
		$i = ($i / 2);
	}
	return $m;
}

Function Fun6($n)
{
	$i = null;
	$j = null;
	$k = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $n); ++ $j)
	For($k = 0; ($k < $n); ++ $k)
	$m += 1;
	return $m;
}

Function Fun7($n)
{
	$i = null;
	$j = null;
	$k = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $n); ++ $j)
	$m += 1;

	For($i = 0; ($i < $n); ++ $i)
	For($k = 0; ($k < $n); ++ $k)
	$m += 1;
	return $m;
}

Function Fun8($n)
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $n ); ++ $j) // Math->sqrt
	$m += 1;
	return $m;
}

Function Fun9($n)
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = $n; ($i >= 1); $i /= 2)
	For($j = 0; ($j < $i); ++ $j)
	$m += 1;
	return $m;
}

Function Fun10($n) 
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = $i; ($j > 0); -- $j)
	$m += 1;
	return $m;
}

Function Fun11($n) // [int n]
{
	$i = null;
	$j = null;
	$k = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = $i; ($j < $n); ++ $j)
	For($k = ($j + 1); ($k < $n); ++ $k)
	$m += 1;
	return $m;
}

Function Fun12($n) 
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = 0; ($i < $n); ++ $i)
	For($j = 0; ($j < $n); ++ $j)
	$m += 1;
	return $m;
}

Function Fun13($n) 
{
	$i = null;
	$j = null;
	$m = 0;
	For($i = 1; ($i < $n); $i *= 2)
	For($j = 0; ($j < $i); ++ $j)
	$m += 1;

	return $m;
}

echo Fun1(100) . "<br/>";
echo Fun2(100) . "<br/>";
echo Fun3(100) . "<br/>";
echo Fun4(100) . "<br/>";
echo Fun5(100) . "<br/>";
echo Fun6(100) . "<br/>";
echo Fun7(100) . "<br/>";
echo Fun8(100) . "<br/>";
echo Fun9(100) . "<br/>";
echo Fun10(100) . "<br/>";
echo Fun11(100) . "<br/>";
echo Fun12(100) . "<br/>";
echo Fun13(100) . "<br/>";
?>